### PR TITLE
Introduce forbidden-apis to RESTEasy Reactive

### PR DESCRIPTION
**Pull request by geoand**

### CODE REVIEW

File: extensions/resteasy-reactive/jaxrs-client-reactive/deployment/banned-signatures.txt [hunk 1]
```diff
@@ -0,0 +1,2 @@
+@defaultMessage Never use Type#toString() as it's almost always the wrong thing to do. Usually org.jboss.jandex.DotName#toString() is what is needed
+org.jboss.jandex.Type#toString()
```
[the remark this answers is not in the document]
Comment (Ladicek) [reply extensions/resteasy-reactive/jaxrs-client-reactive/deployment/banned-signatures.txt:2]: Also the configuration of the Maven plugin seems always the same, maybe that could be put into the Quarkus build-parent (into `<pluginManagement>`, and modules that want the check will just declare the plugin itself)?

Independent project will have to repeat the plugin configuration as well as the signature files, I think, but still better than present situation.

[the remark this answers is not in the document]
Comment (geoand) [reply extensions/resteasy-reactive/jaxrs-client-reactive/deployment/banned-signatures.txt:2]: I think both propositions make sense.

File: core/deployment/banned-signatures.txt [hunk 2]
```diff
@@ -1,5 +0,0 @@
-@defaultMessage Don't use Maven classes. They won't be available when using Gradle.
-org.apache.maven.**
-org.codehaus.plexus.**
```
Comment (Ladicek) [inline core/deployment/banned-signatures.txt:3]: This ban of Maven classes is now gone?

Comment (geoand) [reply core/deployment/banned-signatures.txt:3]: Ouch... bad copy paste

Comment (geoand) [reply core/deployment/banned-signatures.txt]: Should be fixed now, thanks!

**Review by Ladicek — APPROVED** [review]
LGTM, with the caveat that `maven.multiModuleProjectDirectory` is apparently considered internal and should not be relied upon.

See:

- https://issues.apache.org/jira/browse/MNG-6589
- https://issues.apache.org/jira/browse/MNG-7038
- https://github.com/apache/maven/pull/840#issuecomment-1286627143

**Pull request merged**